### PR TITLE
[Bug] [MASTER-9811]fix cmd param to overwrite global param when executing complement

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
## Purpose of the pull request

close[ [#9811](https://github.com/apache/dolphinscheduler/issues/9811)](https://github.com/apache/dolphinscheduler/issues/9811)

## Brief change log

close[ [#9811](https://github.com/apache/dolphinscheduler/issues/9811)](https://github.com/apache/dolphinscheduler/issues/9811)
## Verify this pull request

have test locally

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 1]
```diff
@@ -1953,4 +1958,36 @@ private boolean isNewProcessInstance() {
         }
     }
 
+    private void setGlobalParamIfCommanded(ProcessDefinition processDefinition, Map<String, String> cmdParam) {
+        // get start params from command param
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1962]: I don't see the whole code, It's better to check the `cmdParam` in the begin, if the `cmdParam` is null, you can directly return. You can simplify the logic.

Comment (Tianqi-Dotes) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1962]: Map<String, String> cmdParam = JSONUtils.toMap(processInstance.getCommandParam());
            if (cmdParam != null && cmdParam.containsKey(CMDPARAM_COMPLEMENT_DATA_START_DATE)) {

Comment (Tianqi-Dotes) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1962]: hi ,
the code already have judged null.
yours

Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1962]: Yes, I mean you can just check once in the beginning.

Comment (Tianqi-Dotes) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1962]: have done , thanx

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**